install/error-handling.sh

```diff
@@ -1,10 +1,34 @@
 echo "${_group}Setting up error handling ..."
 
+export SENTRY_DSN='https://afa81b4b7c634e318b0ec138abb2645f@self-hosted.getsentry.net/2'
+export SENTRY_ORG=sentry
+export SENTRY_PROJECT=dogfooding
+
+if [[ -f .reporterrors ]]; then
+  if [[ "$(cat .reporterrors)" == "yes" ]]; then
+    export REPORT_ERRORS=1
+  else
+    export REPORT_ERRORS=0
+  fi
+else
+  echo "Would you like to opt-in to error monitoring for the Sentry installer?"
+  echo "This helps us catch and fix errors when installing Sentry."
+  echo "We may retain your OS username, IP address, and installer log, for 30 days."
+  echo "Your information is solely used for error monitoring of the installer."
+  echo "You may change your preference at any time by deleting the '.reporterrors' file."
+  select yn in "Yes" "No"; do
+      case $yn in
+          Yes ) export REPORT_ERRORS=1; echo "yes" > .reporterrors; break;;
+          No ) export REPORT_ERRORS=0; echo "no" > .reporterrors; break;;
+      esac
+  done
+fi
+
 # Courtesy of https://stackoverflow.com/a/2183063/90297
 trap_with_arg() {
   func="$1" ; shift
   for sig ; do
-    trap "$func $sig "'$LINENO' "$sig"
```

Comment:
is this because our error monitoring will be more detailed so this is not needed?

Reply:
Yes, the stack trace includes the lineno for each part of the stack up to where the error occured.